scripts/prepare_jailbreaks_data.py

```diff
@@ -0,0 +1,326 @@
+        return {
+            'text': example['vanilla'],
+            'label': example['data_type']
+        }
+
+    def map_to_adversarial_example(example):
+        return {
+            'text': example['adversarial'],
+            'label': example['data_type']
+        }
+
+    # Filter and process the dataset
+    def process_dataset(dataset, valid_fn, map_fn, is_eval: bool = False):
+        # Step 1: Filter out invalid examples
+        filtered_dataset = dataset.filter(valid_fn)
+        
+        # Step 2: Map to the desired structure
+        processed_dataset = filtered_dataset.map(
+            map_fn,
+            remove_columns=['vanilla', 'adversarial', 'completion', 'data_type'] if not is_eval else ['adversarial', 'data_type'],  # Drop unnecessary columns
+        )
+        
+        return processed_dataset
+
+    vanilla_train_pool = process_dataset(jailbreak_train['train'], is_valid_vanilla_example, map_to_vanilla_example)
+    adversarial_train_pool = process_dataset(jailbreak_train['train'], is_valid_adversarial_example, map_to_adversarial_example)
+
+
+    wildjailbreak_test = process_dataset(jailbreak_eval, is_valid_adversarial_example, map_to_adversarial_example, is_eval=True)
+
+    alert_vanilla = load_dataset("Babelscape/ALERT", "alert")
+    alert_adversarial = load_dataset("Babelscape/ALERT", "alert_adversarial")
+
+
+    alert_vanilla = alert_vanilla.rename_column("prompt", "text")
+
+    alert_adversarial = alert_adversarial.rename_column("prompt", "text")
+
+
+    # Get unique categories and their counts for vanilla dataset
+    vanilla_categories = alert_vanilla["test"].unique("category")
+    vanilla_category_counts = {cat: len(alert_vanilla["test"].filter(lambda x: x["category"] == cat)) for cat in vanilla_categories}
+
+    # Initialize empty lists for train and test indices
+    vanilla_train_indices = []
+    vanilla_test_indices = []
+
+    # Split each category roughly in half while maintaining proportions
+    for category in vanilla_categories:
+        category_indices = [i for i, x in enumerate(alert_vanilla["test"]) if x["category"] == category]
+        half_size = len(category_indices) // 2
+        
+        # Shuffle indices for this category
+        shuffled_indices = category_indices.copy()
+        random.seed(42)
+        random.shuffle(shuffled_indices)
+        
+        vanilla_train_indices.extend(shuffled_indices[:half_size])
+        vanilla_test_indices.extend(shuffled_indices[half_size:half_size*2])
+
+    # Create stratified train/test splits for vanilla
+    alert_vanilla_train = alert_vanilla["test"].select(vanilla_train_indices)
+    alert_vanilla_test = alert_vanilla["test"].select(vanilla_test_indices)
+
+    # Remove any overlapping examples between train and test based on text
+    alert_vanilla_test_texts = set(alert_vanilla_test["text"])
+    alert_vanilla_train = alert_vanilla_train.filter(lambda x: x["text"] not in alert_vanilla_test_texts)
```

Comment:
> have smaller sets than what you wanted/could achieve
The deduplication is really anecdotal, I ran a few random seeds and it removed a dozen or so documents at most, seemingly because occasionally alert re-used the same prompt. I'm keeping it as-is as getting signal on ALERT was decently strange and data analysis showed it'd take more time than I'd like to -- esp. as it's not an industry standard benchmark (yet?)